src/Control/Middleware/RewriteHashLinksMiddleware.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace SilverStripe\Control\Middleware;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Middleware\HTTPMiddleware;
+use SilverStripe\Core\Convert;
+use SilverStripe\Core\Config\Configurable;
+
+class RewriteHashLinksMiddleware implements HTTPMiddleware
+{
+    use Configurable;
+
+    /**
+     * Mime types to be handled by this middleware
+     * @var string
+     * @config SilverStripe\Control\Middleware\RewriteHashLinksMiddleware.handled_mime_types
+     */
+    private static $handled_mime_types = [
+        'text/html',
+        'application/xhtml+xml',
+    ];
+
+
+    /**
+     * Rewrites hash links in html responses
+     * @param HTTPRequest $request
+     * @param callable $delegate
+     * @return \SilverStripe\Control\HTTPResponse
+     */
+    public function process(HTTPRequest $request, callable $delegate)
+    {
+        /** @var \SilverStripe\Control\HTTPResponse $response **/
+        $response = $delegate($request);
+
+
+        $contentType = explode(';', $response->getHeader('content-type'));
+        $contentType = strtolower(trim(array_shift($contentType)));
+        if (in_array($contentType, $this->config()->handled_mime_types)) {
+            $body = $response->getBody();
+
+            if (strpos($body, '<base') !== false) {
```

Comment:
```suggestion
            if (strpos($body, '<base') === false) {
               return;
            }
```
Then unindent the lines below

Reply:
I don't think this would be correct if there is no base tag there isn't really a need to do hash rewriting. In large part this code is a lift and move from SSViewer which is checking `!== false` https://github.com/silverstripe/silverstripe-framework/blob/4/src/View/SSViewer.php#L684-L694